src/test/mir-opt/simplify-arm-identity/32bit/rustc.main.SimplifyArmIdentity.diff

```diff
@@ -28,16 +28,34 @@
                                            // + literal: Const { ty: u8, val: Value(Scalar(0x00)) }
           discriminant(_1) = 0;            // scope 0 at $DIR/simplify-arm-identity.rs:18:18: 18:29
           StorageLive(_2);                 // scope 1 at $DIR/simplify-arm-identity.rs:19:18: 22:6
-          _3 = const 0isize;               // scope 1 at $DIR/simplify-arm-identity.rs:20:9: 20:20
+          _3 = discriminant(_1);           // scope 1 at $DIR/simplify-arm-identity.rs:20:9: 20:20
```

Comment:
cc @wesleywiser I think we need to adjust the simplify pass to handle deaggregated MIR